scheduler/scheduler_sysbatch_test.go

```diff
@@ -910,8 +965,10 @@ func TestSysBatch_JobConstraint_AddNode(t *testing.T) {
 	require.Nil(t, h.Process(NewSysBatchScheduler, eval3))
 	require.Equal(t, "complete", h.Evals[2].Status)
 
-	// Ensure no failed TG allocs
-	require.Equal(t, 0, len(h.Evals[2].FailedTGAllocs))
+	// Ensure `groupA` fails to be placed due to its constraint, but `groupB` doesn't
+	require.Equal(t, 1, len(h.Evals[2].FailedTGAllocs))
```

Comment:
[`require.Len`](https://pkg.go.dev/github.com/stretchr/testify/require#Len) is a handy test for this; on failure, it prints the map so we can inspect state of map to understand how the count differed.

```suggestion
	require.Len(t, h.Evals[2].FailedTGAllocs, 1)
```